go/chat/boxer.go

```diff
@@ -908,35 +917,64 @@ func (b *Boxer) BoxMessage(ctx context.Context, msg chat1.MessagePlaintext,
 		return nil, NewBoxingError(msg, false)
 	}
 
-	// CORE-4540: MerkleRoot will be in signed header.
-	//            It is useless to put it in the unsigned header in the meantime.
-	// merkleRoot, err := b.latestMerkleRoot()
-	// if err != nil {
-	// 	return nil, NewBoxingError(err.Error(), false)
-	// }
-	// msg.ClientHeader.MerkleRoot = merkleRoot
+	err = b.attachMerkleRoot(ctx, &msg)
+	if err != nil {
+		return nil, err
+	}
 
 	if len(msg.ClientHeader.TlfName) == 0 {
 		msg := fmt.Sprintf("blank TLF name received: original: %s canonical: %s", tlfName,
 			msg.ClientHeader.TlfName)
 		return nil, NewBoxingError(msg, true)
 	}
 
-	boxed, err := b.box(msg, encryptionKey, signingKeyPair, b.boxWithVersion)
+	boxed, err := b.box(ctx, msg, encryptionKey, signingKeyPair, b.boxWithVersion)
 	if err != nil {
 		return nil, NewBoxingError(err.Error(), true)
 	}
 
 	return boxed, nil
 }
 
-func (b *Boxer) box(messagePlaintext chat1.MessagePlaintext, encryptionKey types.CryptKey,
+// Attach a merkle root to the message to send.
+// Modifies msg.
+// For MessageBoxedV1 makes sure there is no MR.
+// For MessageBoxedV2 attaches a (possibly quite stale) MR.
```

Comment:
If we update that duration constant to something tighter, is the `(possibly quite stale)` comment here still true?